src/js/components/user-management/twofactorauthsetup.js

```diff
@@ -18,12 +18,26 @@ export class TwoFactorAuthSetup extends React.Component {
     const self = this;
     self.state = { validated2fa: false, validating2fa: false };
     self.props.saveGlobalSettings({ '2fa': 'enabled' });
+    self.onUnload = self.onUnload.bind(self);
+  }
+
+  componentDidMount() {
+    window.addEventListener('beforeunload', this.onUnload);
   }
 
   componentWillUnmount() {
-    if (!this.state.validated2fa) {
+    if (!this.state.validated2fa && this.props.qrImage) {
```

Comment:
isnt that so: `this.props.qrImage` is not null only when qr is loaded? so, it is possible to click the switch to enable 2fa, and close the browser window before qrImage is fully loaded, and fool this? maybe use `qrExpanded` in here?

Reply:
well before the 2fa is fully loaded it won't be in the store - rendering is independent of this... + yes this is only relevant if `qrExpanded`...